plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,144 @@
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    public static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    public static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+
+    private final TypeManager typeManager;
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public DeltaLakeTableChangesFunction(TypeManager typeManager, DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(START_VERSION_ARGUMENT).type(BIGINT).defaultValue(null).build()),
+                GENERIC_TABLE);
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, CatalogHandle catalogHandle, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument schemaNameArgument = (ScalarArgument) arguments.get(TABLE_SCHEMA_NAME_ARGUMENT);
+        requireNonNull(schemaNameArgument.getValue(), "schemaName value for function table_changes() is null");
+        String schemaName = ((Slice) schemaNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        requireNonNull(tableNameArgument.getValue(), "tableName value for function table_changes() is null");
+        String tableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument startVersionArgument = (ScalarArgument) arguments.get(START_VERSION_ARGUMENT);
+        Long startVersion = (Long) startVersionArgument.getValue();
+        if (startVersion == null) {
+            startVersion = 0L;
+        }
+        else {
+            startVersion++; // to ensure that the startVersion is exclusive
+        }
+        checkArgument(startVersion >= 0, "startVersion value for function table_changes() must not be negative");
+
+        MetadataEntry metadata = getTableMetadata(session, schemaName, tableName);
+
+        ImmutableList.Builder<Descriptor.Field> outputFields = ImmutableList.builder();
+        List<DeltaLakeColumnMetadata> deltaLakeColumnMetadata = extractSchema(metadata, typeManager);
```

Comment:
which method exactly ? 
`io.trino.plugin.deltalake.DeltaLakeMetadata#getColumnHandles` - returns map instead of list and contains `path`, 
`fileSize` and `fileModifiedTime` columns that I don't need.
`io.trino.plugin.deltalake.DeltaLakeMetadata#getColumns` is private and contains the same not needed columns
I'd gladly use existing implementation but I don't see it.
I can also add a method to do exactly what I need there if this is what you thought about